api/gateway/gateway.proto

```diff
@@ -9,13 +9,28 @@ package gateway;
 
 option go_package = "github.com/TheThingsNetwork/ttn/api/gateway";
 
-message GPSMetadata {
+message LocationMetadata {
   // Time in Unix nanoseconds
   int64   time       = 1;
 
   float   latitude   = 2;
   float   longitude  = 3;
   int32   altitude   = 4;
+
+  LocationSource source = 5;
+
+  enum LocationSource {
+    // The source of the location is not known or not set
+    UNKNOWN  = 0;
+    // The location is determined by GPS
+    GPS      = 1;
+    // The location is fixed by configuration
+    CONFIG   = 2;
+    // The location is set in and updated from a registry
+    REGISTRY = 3;
+    // The location is estimated
```

Comment:
In that case I think we should leave it out for now, and add specific Enum values for different estimation techniques when we implement them. Unless we can come up with a decent list of techniques we expect to use.